newsroom/agenda/agenda.py

```diff
@@ -386,91 +386,131 @@ def _set_event_date_range(search):
 
     should = []
 
+    # Scenario 1: Default query with only date_from
     if date_from and not date_to:
-        # Filter from a particular date onwards
-        should = gen_date_range_filter("dates.end", "gte", search.args.get("date_from"), date_from)
+        should.extend(
+            [
+                # Events that start after or on date_from
+                {
+                    "bool": {
+                        "should": [
+                            {"range": {"dates.start": {"gte": date_from}}},
+                            {"range": {"dates.end": {"gte": date_from}}},
+                        ],
+                        "must_not": {"term": {"dates.all_day": True}},
+                    }
+                },
+                # All-day events starting or ending after date_from
+                {
+                    "bool": {
+                        "should": [
+                            {"range": {"dates.start": {"gte": date_from}}},
+                            {"range": {"dates.end": {"gte": date_from}}},
+                        ],
+                        "filter": {"term": {"dates.all_day": True}},
+                    }
+                },
+            ]
+        )
+
+    # Scenario 2 and 3: Query with specific filters
     elif not date_from and date_to:
-        # Filter up to a particular date
-        should = gen_date_range_filter("dates.end", "lte", search.args.get("date_to"), date_to)
+        # Apply date range filters for next_7_days, next_30_days, etc.
```

Comment:
it's kind of confusing that it's checking if `date_to` is set and then it's not used